### ISSUE TITLE
SSL3_GET_CLIENT_HELLO:no shared cipher after updating SSL certificate

### ISSUE THREAD

**Issue by jbeyer05** [puma/puma#1339]
After updating my SSL certificate, any requests to Puma produce the following error messages from Puma's side:

2017-06-19 12:30:19 -0400: SSL error, peer: 73.10.182.229, peer cert: , #<Puma::MiniSSL::SSLError: OpenSSL error: error:1408A0C1:SSL routines:SSL3_GET_CLIENT_HELLO:no shared cipher - 336109761>
2017-06-19 12:30:19 -0400: SSL error, peer: 73.10.182.229, peer cert: , #<Puma::MiniSSL::SSLError: OpenSSL error: error:1408A0C1:SSL routines:SSL3_GET_CLIENT_HELLO:no shared cipher - 336109761>
2017-06-19 12:30:19 -0400: SSL error, peer: 73.10.182.229, peer cert: , #<Puma::MiniSSL::SSLError: OpenSSL error: error:1408A0C1:SSL routines:SSL3_GET_CLIENT_HELLO:no shared cipher - 336109761>

Both certificates came from the same issuer.  Both seem identical in setup, with the one exception being that the new certificate is for a wildcard domain.

Thanks in advance!

**Comment by rickpeyton** [comment]
I have the same issue.

```
web_1  | 2017-07-24 22:58:10 +0000: SSL error, peer: 69.58.181.15, peer cert: , #<Puma::MiniSSL::SSLError: OpenSSL error: error:140760FC:SSL routines:SSL23_GET_CLIENT_HELLO:unknown protocol - 336027900>
web_1  | 2017-07-24 22:58:18 +0000: SSL error, peer: 69.58.181.15, peer cert: , #<Puma::MiniSSL::SSLError: OpenSSL error: error:1408A10B:SSL routines:SSL3_GET_CLIENT_HELLO:wrong version number - 336109835>
web_1  | 2017-07-24 22:58:26 +0000: SSL error, peer: 69.58.181.15, peer cert: , #<Puma::MiniSSL::SSLError: OpenSSL error: error:1408A0C1:SSL routines:SSL3_GET_CLIENT_HELLO:no shared cipher - 336109761>
web_1  | 2017-07-24 22:58:33 +0000: SSL error, peer: 69.58.181.15, peer cert: , #<Puma::MiniSSL::SSLError: OpenSSL error: error:1408A0C1:SSL routines:SSL3_GET_CLIENT_HELLO:no shared cipher - 336109761>
web_1  | 2017-07-24 22:58:41 +0000: SSL error, peer: 69.58.181.15, peer cert: , #<Puma::MiniSSL::SSLError: OpenSSL error: error:1408A0C1:SSL routines:SSL3_GET_CLIENT_HELLO:no shared cipher - 336109761>
web_1  | 2017-07-24 22:58:41 +0000: SSL error, peer: 69.58.181.15, peer cert: , #<Puma::MiniSSL::SSLError: OpenSSL error: error:140A1175:SSL routines:SSL_BYTES_TO_CIPHER_LIST:inappropriate fallback - 336204149>
web_1  | 2017-07-24 22:58:48 +0000: SSL error, peer: 69.58.181.15, peer cert: , #<Puma::MiniSSL::SSLError: OpenSSL error: error:1408A10B:SSL routines:SSL3_GET_CLIENT_HELLO:wrong version number - 336109835>
web_1  | 2017-07-24 22:58:57 +0000: SSL error, peer: 69.58.181.15, peer cert: , #<Puma::MiniSSL::SSLError: OpenSSL error: error:1408F119:SSL routines:SSL3_GET_RECORD:decryption failed or bad record mac - 336130329>
web_1  | 2017-07-24 22:58:57 +0000: SSL error, peer: 69.58.181.15, peer cert: , #<Puma::MiniSSL::SSLError: OpenSSL error: error:1408F119:SSL routines:SSL3_GET_RECORD:decryption failed or bad record mac - 336130329>
web_1  | 2017-07-24 22:58:57 +0000: SSL error, peer: 69.58.181.15, peer cert: , #<Puma::MiniSSL::SSLError: OpenSSL error: error:1408F119:SSL routines:SSL3_GET_RECORD:decryption failed or bad record mac - 336130329>
web_1  | 2017-07-24 22:58:57 +0000: SSL error, peer: 69.58.181.15, peer cert: , #<Puma::MiniSSL::SSLError: OpenSSL error: error:1408F119:SSL routines:SSL3_GET_RECORD:decryption failed or bad record mac - 336130329>
```

Looks fine in the browser and https://cryptoreport.websecurity.symantec.com/checker/ does not report any problems, but my logs are blowing up with the error above.

Any thoughts? My Google-fu failed me.

**Comment by jbeyer05** [comment]
@rickpeyton I resorted to using nginx as a reverse proxy.  Wish I had another solution.

**Comment by rickpeyton** [comment]
Thanks @jbeyer05 I was hoping to avoid nginx for this little project, but I will do that too. Thanks again!

**Comment by Bodacious** [comment]
Having this same issue today trying to serve a Rails app locally with SSL. Any ideas?

**Comment by Fortunate-MAN** [comment]
I'm having the same issue with a Rails app I'm trying to serve publicly on a domain; doesn't work locally either. Any help?

**Comment by Fortunate-MAN** [comment]
I ended up using Passenger to support SSL; works beautifully.

**Comment by hseljenes** [comment]
This usually means there was an issue loading your crt or key file. In my case I accidentally mapped them into my containers as empty directories instead of files. Double check the user running puma can read your files and that the paths are correct.

**Comment by caljess599** [comment]
I had a similiar issue to @hseljenes. Puma could read the files, but there was something wrong with them. Finally, I stumbled on advice to test the cert installation with `openssl s_server -key /path/to/key -cert path/to/cert` ; that command diagnosed it immediately as a `key values mismatch` (Then I learned how to see how the key/cert didn't match for myself [here](https://knowledge.digicert.com/solution/SO17751.html).) 

I don't work too often with certs, and when I bought my cert, the instructions were lacking, and so I didn't know to insert the cert at the beginning of the "bundle" file sent to me that provides the authority chain. So it wasn't a proper cert until I did that. Live and learn...
